### DISCUSSION TITLE
Warning: GPDSC generation will be soon discontinued

### DISCUSSION THREAD

**Discussion by FrankBau** [ussserrr/stm32pio#113] [Core]
With STM32CubeMX 6.4.0 (Nov. 2021) I get a warning on code generation:

  - The support of the “Tool chain / IDE” you have selected will be soon discontinued in further STM32CubeMX version.   If you want to benefit from further STM32CubeMX evolutions for your project,   we advise you to target STM32CubeIDE or other IDE.

What can we do?

KnarfB

**Comment by ussserrr** [comment]
I've just investigated this issue and I think we can safely use "STM32CubeIDE" option as well since the output code format seems exactly the same (or at least compatible) and can be successfully built by PlatformIO. I will plan to remove "Incompatible option" warning from stm32pio. At the meantime, current settings can be used and CubeMX actually does not produce such a message when invoked through the CLI (via stm32pio). Thank you for information!

**Reply by FrankBau** [reply]
Thank you for providing the code.
KnarfB
